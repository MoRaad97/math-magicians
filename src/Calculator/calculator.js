import React from 'react';
// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  render() {
    return (
      <>
        <div className="calculator-container">
          <div className="screen"><span className="result">0</span></div>
          <div className="buttons">
            <button type="button" className="normal">AC</button>
            <button type="button" className="normal">+/-</button>
            <button type="button" className="normal">%</button>
            <button type="button" className="normal active"><span>&#247;</span></button>
            <button type="button" className="normal">7</button>
            <button type="button" className="normal">8</button>
            <button type="button" className="normal">9</button>
            <button type="button" className="normal active">x</button>
            <button type="button" className="normal">4</button>
            <button type="button" className="normal">5</button>
            <button type="button" className="normal">6</button>
            <button type="button" className="normal active">-</button>
            <button type="button" className="normal">1</button>
            <button type="button" className="normal">2</button>
            <button type="button" className="normal">3</button>
            <button type="button" className="normal active">+</button>
            <button type="button" className="zero">0</button>
            <button type="button" className="normal">.</button>
            <button type="button" className="normal active">=</button>
          </div>
        </div>
      </>
    );
  }
}

export default Calculator;
