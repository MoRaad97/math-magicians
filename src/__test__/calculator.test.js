import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Components/calculator'

it.only('render the Calculator Component', ()=> {
const tree = renderer.create(<Calculator/>).toJSON();
expect(tree).toMatchSnapshot()
})