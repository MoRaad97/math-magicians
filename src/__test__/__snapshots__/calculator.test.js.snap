// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the Calculator Component 1`] = `
<section
  className="calculator"
>
  <div
    className="my-calc"
  >
    Let's Do Some Math Magic:
  </div>
  <div
    className="calculator-container"
  >
    <div
      className="screen"
    >
      <span
        className="result"
      >
        0
      </span>
    </div>
    <div
      className="buttons"
    >
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="normal active"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="normal active"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="normal active"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="normal active"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="normal"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="normal active"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
